### DISCUSSION TITLE
Update "The Matrix" Demo for ST7789?

### DISCUSSION THREAD

**Discussion by phillmj** [Bodmer/TFT_eSPI#2219] [General]
Hi Mr. Bodmer, I can't get your cool Matrix example working - I've got a TEENSY 4.0 and using a Waveshare ST7789 screen. I notice some of the commands that fail in the compiler refer to the IL9341 lib. I can't find equivalents.

Do you have any pointers how I could get this working on this screen? I'd also love to be able to specify the font used, as I found a very matrix-like Japanese character set that is supposedly used in the actual film.

Also, I have interacted with the maker of the original version, Eric, and I have gotten his working, but I'd love to see your take.

Thanks for all you do, please take some time out for yourself I don't know how you answer all these questions and write all this fabulous code.

Thanks

Mike

**Comment by Bodmer** [comment]
That example uses the hardware scrolling feature of the ILI9341 so would need adaption for other displays.

You might like to try this library and example instead as it should work on other display types:
https://github.com/0015/TP_Arduino_DigitalRain_Anim

**Comment by phillmj** [comment]
Thanks so much. Yes, I did that already, and did adapt it, and it's working well. I just wanted to see if I could get your example working, because you have a unique take on things.

Thanks again.

Mike

**Comment by Bodmer** [comment]
If you replace:
ILI9341_VSCRDEF
ILI9341_VSCRSADD
In the Matrix example with:
ST7789_VSCRDEF
ST7789_VSCRSADD
It should work.

**Reply by phillmj** [reply]
Thanks so much! That did the trick!!!! :) :) :)
